BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BTConnectToThread.java

```diff
@@ -40,21 +54,46 @@ public BTConnectToThread(BtConnectToCallback Callback, BluetoothDevice device, U
     }
     public void run() {
         printe_line("Starting to connect");
-        if(mSocket != null && callback != null) {
+        if (mSocket == null || callback == null) {
+            return;
```

Comment:
Don't we need to call the callback and let them know that no connection is going to happen? Ever?